jaxrs/src/main/java/com/facebook/airlift/jaxrs/AuthorizationFilter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.jaxrs;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+@Provider
+public class AuthorizationFilter
+        implements ContainerRequestFilter
+{
+    private final List<Authorizer> authorizers;
+
+    @Context
+    private ResourceInfo resourceInfo;
+
+    @Inject
+    public AuthorizationFilter(List<Authorizer> authorizers)
+    {
+        this.authorizers = ImmutableList.copyOf(requireNonNull(authorizers, "authorizers is null"));
+    }
```

Comment:
If someone enables the authorization filter - they should be passing an authorizer